lib/presentation/pages/create_trip_page/create_trip_basic_info_page.dart

```diff
@@ -16,81 +24,159 @@ import 'package:trip_tally/presentation/widgets/m3_widgets/navigation_app_bar.da
 
 @RoutePage()
-class CreateTripBasicInfoPage extends StatefulWidget {
-  const CreateTripBasicInfoPage({super.key, this.cubit});
+class CreateTripBasicInfoPage extends StatelessWidget {
+  const CreateTripBasicInfoPage({
+    this.bloc,
+    super.key,
+    this.cubit,
+  });
 
   final OsmSuggestionsCubit? cubit;
-
-  @override
-  CreateTripBasicInfoPageState createState() => CreateTripBasicInfoPageState();
-}
-
-class CreateTripBasicInfoPageState extends State<CreateTripBasicInfoPage> {
-  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final CreateTripBloc? bloc;
 
   @override
   Widget build(BuildContext context) {
-    return BlocProvider(
-      create: (context) => widget.cubit ?? getIt<OsmSuggestionsCubit>(),
-      child: BlocListener<OsmSuggestionsCubit, OsmSuggestionsState>(
-        listener: (context, state) => state.whenOrNull(
-          error: (error) => showSnackBar(context, error.errorText(context)),
+    return MultiBlocProvider(
+      providers: [
+        BlocProvider(
+          create: (context) => cubit ?? getIt<OsmSuggestionsCubit>(),
+          child: BlocListener<OsmSuggestionsCubit, OsmSuggestionsState>(
+            listener: (context, state) => state.whenOrNull(
+              error: (error) => showSnackBar(context, error.errorText(context)),
+            ),
+          ),
         ),
-        child: Scaffold(
-          floatingActionButton: ProceedFloatingActionButton(
-            onPressed: () {
-              _formKey.currentState?.validate();
-            },
-          ).animate().scale(delay: 400.ms),
-          appBar: NavigationAppBar(title: context.tr.createTripPage_titleBasicInfo),
-          body: _Body(formKey: _formKey),
+        BlocProvider(
+          create: (context) => bloc ?? getIt<CreateTripBloc>(),
+          child: BlocListener<CreateTripBloc, CreateTripState>(
+            listener: (context, state) => state.whenOrNull(
+              failure: (error) => showSnackBar(
+                context,
+                error.errorText(context),
+              ),
+              success: () => context.router.push(const HomeRoute()),
+            ),
+          ),
         ),
-      ),
+      ],
+      child: const _Body(),
     );
   }
 }
 
 class _Body extends StatefulWidget {
-  const _Body({required this.formKey});
-
-  final GlobalKey<FormState> formKey;
+  const _Body();
 
   @override
   State<_Body> createState() => _BodyState();
 }
 
 class _BodyState extends State<_Body> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
   late final TextEditingController _currencyController;
   late final TextEditingController _budgetController;
+  late final TextEditingController _cityNameController;
+  late final TextEditingController _countryCode;
+  DateTime? _startDate;
+  DateTime? _endDate;
+  TransportType _transportType = TransportType.flight;
+  late bool isPickedDate;
```

Comment:
you can remove it and in the build method create a `isPickedDate` boolean which will check if _startDate and _endDate are not null